metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -0,0 +1,274 @@
+        return "mongo"
+    if sqlalchemy_uri.startswith("hive"):
+        return "hive"
+    return "external"
+
+
+class SupersetSource(Source):
+    config: SupersetConfig
+    report: SourceReport
+    env = "PROD"
+    platform = "superset"
+
+    def __hash__(self):
+        return 0
+
+    def __init__(self, ctx: PipelineContext, config: SupersetConfig):
+        super().__init__(ctx)
+        self.config = config
+        self.report = SourceReport()
+
+        login_response = requests.post(
+            f"{self.config.connect_uri}/api/v1/security/login",
+            None,
+            {
+                "username": self.config.username,
+                "password": self.config.password,
+                "refresh": True,
+                "provider": self.config.provider,
+            },
+        )
+
+        self.access_token = login_response.json()["access_token"]
+
+        self.session = requests.Session()
+        self.session.headers.update(
+            {
+                "Authorization": f"Bearer {self.access_token}",
+                "Content-Type": "application/json",
+                "Accept": "*/*",
+            }
+        )
+
+        # Test the connection
+        test_response = self.session.get(f"{self.config.connect_uri}/api/v1/database")
+        if test_response.status_code == 200:
+            pass
+            # TODO(Gabe): how should we message about this error?
+
+    @classmethod
+    def create(cls, config_dict: dict, ctx: PipelineContext) -> Source:
+        config = SupersetConfig.parse_obj(config_dict)
+        return cls(ctx, config)
+
+    @lru_cache(maxsize=None)
+    def get_platform_from_database_id(self, database_id):
+        database_response = self.session.get(
+            f"{self.config.connect_uri}/api/v1/database/{database_id}"
+        ).json()
+        sqlalchemy_uri = database_response.get("result", {}).get("sqlalchemy_uri")
+        return get_platform_from_sqlalchemy_uri(sqlalchemy_uri)
+
+    @lru_cache(maxsize=None)
+    def get_datasource_urn_from_id(self, datasource_id):
+        dataset_response = self.session.get(
+            f"{self.config.connect_uri}/api/v1/dataset/{datasource_id}"
+        ).json()
+        schema_name = dataset_response.get("result", {}).get("schema")
+        table_name = dataset_response.get("result", {}).get("table_name")
+        database_id = dataset_response.get("result", {}).get("database", {}).get("id")
+        database_name = (
+            dataset_response.get("result", {}).get("database", {}).get("database_name")
+        )
+
+        if database_id and table_name:
+            platform = self.get_platform_from_database_id(database_id)
+            platform_urn = f"urn:li:dataPlatform:{platform}"
+            dataset_urn = (
+                f"urn:li:dataset:("
+                f"{platform_urn},{database_name + '.' if database_name else ''}"
+                f"{schema_name + '.' if schema_name else ''}"
+                f"{table_name},{self.env})"
```

Comment:
definitely not necessary for this PR, but we have mce_builders.py which lets you do `builder.make_dataset_urn(platform, name, env)` - my eventual goal is to put all the URN construction stuff there